File: app/src/main/java/com/alex/miet/mobile/data/news/NewsRepositoryImpl.java
package com.alex.miet.mobile.data.news;

import java.util.List;

import com.alex.miet.mobile.NewsModel;
import com.alex.miet.mobile.NewsModelDao;

/**
 * Specific {@link NewsRepository} implementation
 */
public class NewsRepositoryImpl implements NewsRepository {

    private final NewsModelDao dao;

    public NewsRepositoryImpl(NewsModelDao dao) {
        this.dao = dao;
    }

    @Override
    public void save(NewsModel entity) {
        dao.save(entity);
    }

    @Override
    public void saveAll(Iterable<NewsModel> entities) {
        dao.saveInTx(entities);
    }

    @Override
    public List<NewsModel> getAll() {
        return dao.loadAll();
    }

    @Override
    public void update(NewsModel entity) {
        dao.update(entity);
    }

    @Override
    public void updateAll(Iterable<NewsModel> entities) {
        dao.updateInTx(entities);
    }

    @Override
    public void delete(NewsModel entity) {
        dao.delete(entity);
    }

    @Override
    public void deleteAll() {
        dao.deleteAll();
    }
}

File: app/src/main/java/com/alex/miet/mobile/data/news/NewsRepositoryImpl.kt
package com.alex.miet.mobile.data.news

import com.alex.miet.mobile.daos.NewsDao
import com.alex.miet.mobile.entities.NewsItem
import io.reactivex.Maybe

/**
 * Specific [NewsRepository] implementation
 */
class NewsRepositoryImpl(private val dao: NewsDao) : NewsRepository {

    override fun save(entity: NewsItem) {
        dao.insert(entity)
    }

    override fun saveAll(entities: List<NewsItem>) {
        dao.insert(entities)
    }

    override fun getAll(): Maybe<List<NewsItem>> {
        return dao.loadAll()
    }

    override fun update(entity: NewsItem) {
        dao.update(entity)
    }

    override fun updateAll(entities: List<NewsItem>) {
        dao.update(entities)
    }

    override fun delete(entity: NewsItem) {
        dao.delete(entity)
    }

    override fun deleteAll() {
        dao.deleteAll()
    }
}
